components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,495 @@
+		this._addEventListeners = this._addEventListeners.bind(this);
+		this._chomp = this._chomp.bind(this);
+		this._getLayoutItems = this._getLayoutItems.bind(this);
+		this._handleSlotChange = this._handleSlotChange.bind(this);
+		this._getOverflowMenu = this._getOverflowMenu.bind(this);
+
+		this._throttledResize = throttle(this._handleResize, 15);
+
+		this._overflowHidden = this._overflowHidden ?? false;
+		this.autoShow = this.autoShow ?? false;
+		this.maxToShow = this.maxToShow ?? -1;
+		this.minToShow = this.minToShow ?? 1;
+		this.openerStyle = this.minToShow ?? OPENER_STYLE.DEFAULT;
+		this._mini = this.openerType === OPENER_TYPE.ICON;
+
+		this._refId = 0;
+		this._slotItems = [];
+		this._overflowItems = [];
+	}
+	connectedCallback() {
+		super.connectedCallback();
+		this._addEventListeners();
+	}
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._removeEventListeners();
+	}
+	async firstUpdated() {
+		super.firstUpdated();
+
+		// selected elements
+		this._buttonSlot = this.shadowRoot.querySelector('.item-container');
+		this._buttonSlot.addEventListener('slotchange', this._handleSlotChange);
+
+		this._container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+
+		this._autoShowClass = 'd2l-overflow-group-show';
+		this._autoNoShowClass = 'd2l-overflow-group-no-show';
+
+		// get the items from the button slot
+		this._slotItems = this._getSlotItems();
+		// convert them to layout items (calculate widths)
+		this._layoutItems = this._getLayoutItems(this._slotItems);
+		// convert to dropdown items (for overflow menu)
+		this._dropdownItems = this._slotItems.map((node) => this._convertToDropdownItem(node));
+
+		if (this.autoShow) {
+			this._autoDetectBoundaries(this._slotItems);
+		}
+
+		this._availableWidth = this._container.clientWidth;
+	}
+	render() {
+		let overflowMenu;
+		if (!this._overflowMenuHidden) {
+			overflowMenu = this._getOverflowMenu();
+		}
+
+		this._slotItems.forEach((element, index) => {
+			if (!this._overflowMenuHidden && index >= this._chompIndex) {
+				element.setAttribute('data-is-chomped', true);
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		return html`
+			<div class="d2l-overflow-group-container">
+				<slot class="item-container"></slot>
+				${overflowMenu}
+			</div>
+		`;
+	}
+	update(changedProperties) {
+		super.update(changedProperties);
+		if (changedProperties.get('autoShow')) {
+			this._autoDetectBoundaries(this._getLayoutItems());
+		}
+
+		if (changedProperties.get('minToShow')
+			|| changedProperties.get('maxToShow')) {
+			this._chomp();
+		}
+
+		// slight hack, overflowMenu isnt being rendered initially so wait until update
+		// to calculate its width
+		if (!this._overflowMenuWidth) {
+			// this action needs to be deferred until first render of our overflow button
+			requestAnimationFrame(() => {
+				this._chomp();
+			});
+		}
+	}
+	connectedCallack() {
+		this._addEventListeners();
```

Comment:
Since there's just the one, I think `_addEventListeners` could just be pulled into here. (same for remove)